app/main.py

```diff
@@ -1,44 +1,67 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
```

Comment:
The `run_sequence` function may not work as expected. In line 12, you're not actually calling the function, you're just referencing it. To call a coroutine function you should use function with parentheses, like this: `await function()`.

Reply:
Awaitable is not callable